logserver/web-area/put.php

```diff
@@ -0,0 +1,171 @@
+<?php
+// SPDX-FileCopyrightText: 2009 Fermi Research Alliance, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+/*  PHP script to implement file upload via PUT
+    The file content is received via stdin (Apache PUT implementation)
+    A JSON configuration is in logging_config.json, valid parameters are:
+    - secret_key_path (/var/lib/gwms-factory/server-credentials/jwt_secret.key)
+    - secret_key (VG9rZVNlY3JldEtleQo=)
+    - default_file_name (glideinlog.txt)
+    - uri_regex_file_name ()#logging/put.php/(\S+)#)
+    - token_issuer (gethostname())  // Should be the Factory, assuming is the same host by default
+    - upload_dir (uploads/)
+    - upload_dir_unauthorized (uploads_unauthorized/)
+    - require_authentication (True)
+    - verbose (True)  // Print verbose output
+    - debug (False)  // Print debug output
+    The destination file name is retrieved from the URI:
+    - as GET parameter fname, e.g.    ...put.php?fname=NAME
+    - as path, e.g.  ...logging/put.php/NAME
+    The authentication is via JWT
+    - HS256 encoded JWT
+    - iss must be the Factory
+    - the key is stored in '/var/lib/gwms-factory/server-credentials/jwt_secret.key'
+    - aud should be the Web server
+    If requireAuthentication is False, files are uploaded also w/o valid JWT
+*/
+
+// declare(strict_types=1);
+use Firebase\JWT\JWT;
+use Firebase\JWT\Key;
+require_once('../vendor/autoload.php');
+
+// Hardcoded parameters
+$configFile = '../logging_config.json';
+
+// Read and extract configuration
+$config = json_decode(file_get_contents($configFile), true);
+if (! $config) {
+    trigger_error("Empty config ($configFile) - the file may be missing or be invalid JSON", E_USER_WARNING);
+}
+$secretKeyFile = $config['secret_key_path'] ?? '/var/lib/gwms-factory/server-credentials/jwt_secret.key';
+$defaultSecretKey = $config['secret_key'] ?? 'VG9rZVNlY3JldEtleQo=';
```

Comment:
Is it okay/safe to have this secret key defined here (the actual key contents I meant)?

Reply:
removed and protected from empty string